packages/client/components/ReviewRequestToJoinOrgModal.tsx

```diff
@@ -0,0 +1,124 @@
+import graphql from 'babel-plugin-relay/macro'
+import React, {useState} from 'react'
+import {PreloadedQuery, usePreloadedQuery} from 'react-relay'
+import DialogContainer from './DialogContainer'
+import DialogContent from './DialogContent'
+import DialogTitle from './DialogTitle'
+import PrimaryButton from './PrimaryButton'
+import SecondaryButton from './SecondaryButton'
+import {ReviewRequestToJoinOrgModalQuery} from '../__generated__/ReviewRequestToJoinOrgModalQuery.graphql'
+import Checkbox from './Checkbox'
+
+const query = graphql`
+  query ReviewRequestToJoinOrgModalQuery($requestId: ID!) {
+    viewer {
+      domainJoinRequest(requestId: $requestId) {
```

Comment:
+1 I think we should put eligible teams directly on the request and do the server side filtering as you've suggested.